From 051d02b7e2f1ff86af870238252b36ff35f3b3d4 Mon Sep 17 00:00:00 2001
From: Christophe Kerello <christophe.kerello@foss.st.com>
Date: Thu, 6 Jul 2023 10:02:42 +0200
Subject: [PATCH 057/120] memory: stm32-omi: rework stm32_omi_dlyb_stop API

Reset DLYB CR register.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/u-boot.git 2b63d743c012e8c23829ddbb7f428a08db1d80f1 memory: stm32-omi: rework stm32_omi_dlyb_stop API]
Change-Id: Id1fec1c265b6f213fe15a13b054e5d9e38a06c32
Signed-off-by: Christophe Kerello <christophe.kerello@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/u-boot/+/316840
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Tested-by: Christophe KERELLO <christophe.kerello@st.com>
Reviewed-by: Christophe KERELLO <christophe.kerello@st.com>
Reviewed-by: Patrice CHOTARD <patrice.chotard@foss.st.com>
Domain-Review: Christophe KERELLO <christophe.kerello@st.com>
---
 drivers/memory/stm32-omi.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/memory/stm32-omi.c b/drivers/memory/stm32-omi.c
index 17ab6df56a5..c4f94065e00 100644
--- a/drivers/memory/stm32-omi.c
+++ b/drivers/memory/stm32-omi.c
@@ -227,9 +227,9 @@ int stm32_omi_dlyb_stop(struct udevice *dev)
 	int ret;
 
 	/* disable delay block */
-	ret = regmap_update_bits(omi_plat->regmap,
-				 omi_plat->dlyb_base + SYSCFG_DLYBOS_CR,
-				 DLYBOS_CR_EN, 0);
+	ret = regmap_write(omi_plat->regmap,
+			   omi_plat->dlyb_base + SYSCFG_DLYBOS_CR,
+			   0x0);
 
 	if (ret)
 		dev_err(dev, "Error when stopping delay block\n");
-- 
2.47.2

